### PR TITLE
enable milvus as memory backend

### PR DESCRIPTION
**Pull request by chyezh**
### Background

Milvus can be used as a new optional memory backend. It's a open-source software that can be fast deployed by docker or k8s, and there are also commercial solution providers. 

### Changes

- existing function is not modified.
- add Milvus as a new optional memory backend.
- part of document about memory backend is updated. 

### Documentation

New module and code is full documented.
A use guide about new feature is add to README.md.

### Test Plan

- Auto-GPT works properly in milvus memory backend.
- Auto-GPT using local memory works properly too.

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [x] I have thouroughly tested my changes with multiple different prompts.
- [x] I have considered potential risks and mitigations for my changes.
- [x] I have documented my changes clearly and comprehensively.
- [x] I have not snuck in any "extra" small tweaks changes

### CODE REVIEW

**Review by nponeccop — APPROVED** [review]

File: scripts/config.py
Comment (nponeccop) [inline scripts/config.py]: Rollback the whitespace fix. It's painful but it reduces the conflicts. The whitespace PR is already there.

**Review by nponeccop — APPROVED** [review]

File: scripts/config.py
Comment (chyezh) [reply scripts/config.py]: rollback done

**Review by nponeccop — APPROVED** [review]

**Review by nponeccop — APPROVED** [review]

**Comment by chyezh** [comment]
> @chyezh There are conflicts now

**Comment by chyezh** [comment]
@richbeales  conflict fixed, 
Is there anything else need to be changed?
I think that milvus is a convenient way for others to deploy a vector database to retain the memory.

**Review by nponeccop — DISMISSED** [review]
I approve the basic quality of the PR -- linting and focus. But as it is labeled by the maintainers, I think it won't be merged until we merge all small PRs.

**Review by nponeccop — APPROVED** [review]

**Comment by chyezh** [comment]
Conflict fix up.
Update Table of Contents in README.md.
Fix jump feature in Table of Contents.
@nponeccop @richbeales

**Comment by nponeccop** [comment]
@chyezh There are conflicts now

**Review by nponeccop — APPROVED** [review]

**Review by richbeales — APPROVED** [review]

**Comment by isaaclepes** [comment]
I noticed that Milvus, when merged, was placed in scripts/memory/ instead of autogpt/memory/

**Pull request merged**
